src/key.js

```diff
@@ -1380,6 +1336,43 @@ async function wrapKeyObject(secretKeyPacket, secretSubkeyPacket, options) {
   return new Key(packetlist);
 }
 
+/**
+ * Checks if a given certificate or binding signature is revoked
+ * @param  {module:packet/secret_key|
+ *          module:packet/public_key}       primaryKey   The primary key packet
+ * @param  {Object}                         dataToVerify The data to check
+ * @param  {Array<module:packet/signature>} revocations  The revocation signatures to check
+ * @param  {module:packet/signature}        signature    The certificate or signature to check
+ * @param  {module:packet/public_subkey|
+ *          module:packet/secret_subkey|
+ *          module:packet/public_key|
+ *          module:packet/secret_key} key, optional The key packet to check the signature
+ * @param  {Date}                     date          Use the given date instead of the current time
+ * @return {Boolean}                                True if the signature revokes the data
+ */
+async function isDataRevoked(primaryKey, dataToVerify, revocations, signature, key, date=new Date()) {
+  key = key || primaryKey;
+  const normDate = util.normalizeDate(date);
+  const revocationKeyIds = [];
+  await Promise.all(revocations.map(async function(revocationSignature) {
+    if (!(config.revocations_expire && revocationSignature.isExpired(normDate)) &&
+        (revocationSignature.verified || await revocationSignature.verify(key, dataToVerify))) {
+      // TODO get an identifier of the revoked object instead
+      revocationKeyIds.push(revocationSignature.issuerKeyId);
+      return true;
+    }
+    return false;
+  }));
+  // TODO further verify that this is the signature that should be revoked
+  // In particular, if signature.issuerKeyId is a wildcard, any revocation signature will revoke it
```

Comment:
The wild card ID is only defined in the RFC in the context of public-key encrypted session key packets. I don't think we should extrapolate this to all keyIds.